x-pack/plugin/ml/qa/ml-with-security/build.gradle

```diff
@@ -183,6 +183,9 @@ integTest.runner {
     'ml/job_groups/Test put job with id that matches an existing group',
     'ml/job_groups/Test put job with invalid group',
     'ml/ml_info/Test ml info',
+    'ml/pipeline_inference/Test pipeline regression simple',
+    'ml/pipeline_inference/Test pipeline agg referencing a single bucket',
+    'ml/pipeline_inference/Test all fields missing warning',
```

Comment:
None of these are expecting errors in their yaml tests. Why are they part of the blacklist?

Reply:
`MlWithSecurityIT` and `MlWithSecurityUserRoleIT` both pass but `MlWithSecurityInsufficientRoleIT` fails because it expects ml tests to fail if the user does not have one of the ml roles (user, admin). 

In this case you don't need the ml role to run search so the test fails. This definitely opens up a discussion point which needs to be resolved but that can happen outside of this PR. So I hacked a solution to get green

Reply:
Thanks for the explanation :D.

Reply:
Actually looking at `MlWithSecurityUserRoleIT` I think we can do something similar and check which endpoint was invoked by the test.

https://github.com/elastic/elasticsearch/blob/5fa36dad0b8503bef3d91173bb342da555445c26/x-pack/plugin/ml/qa/ml-with-security/src/test/java/org/elasticsearch/smoketest/MlWithSecurityUserRoleIT.java#L46

Reply:
The fix was quite simple I changed `MlWithSecurityInsufficientRoleIT` to check if an ml endpoint was called and only fail if that is the case. 

I pushed that change in c47d42d5c91ef55490abff9683f7a7c0b1bdde0c